iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object.hpp

```diff
@@ -40,6 +40,14 @@ enum class SharedMemoryObjectError
     INTERNAL_LOGIC_FAILURE,
 };
 
+enum class SharedMemoryAllocationError
+{
+    REQUESTED_MEMORY_AFTER_FINALIZED_ALLOCATION,
+    NOT_ENOUGH_MEMORY,
+    REQUESTED_ZERO_SIZED_MEMORY
```

Comment:
Requesting zero memory should be a misuse case. We do not have to change it now as it is a more fundamental issue, but I think we should not mix errors that can happen during correct use and incorrect use. Note that some allocator implementations just treat requesting zero bytes as requesting one byte by convention, but we should not do this IMO.

I have some notion of this in the new error handling and I think the distinction is needed to develop better/safe code.

Nothing to do now.